### ISSUE TITLE
we should allow multiple open offers on the same server as long as the times do not overlap

### ISSUE THREAD

**Issue by tzumainn** [CCI-MOC/flocx-market#137]

**Comment by ljmcgann** [comment]
Are two offers allowed to overlap at endpoints? Like if start_time for the new offer equals end_time of previous, is this ok?

**Comment by tzumainn** [comment]
Good question - I'd say yes, I guess?

**Comment by ljmcgann** [comment]
Ironic nodes don't get reinitialized after being used right? Being that offers only go to one bid, This second offer going to another bid would allow a bidder to be on the hardware at the same time as another bidder for all of one moment. But with network latency and unsynchronized clock nonsense, it seems like some kind of race condition could occur and the new bidder might have slightly more time overlapping with the old bidder and let the old bidder be on the node at the same time as the new one. In fact, do we think there should be some kind of buffer between times to allow for creation?

**Comment by ljmcgann** [comment]
Or am I way overthinking this and stretching to apply my academics into the problem

**Comment by tzumainn** [comment]
Actually, ironic nodes do need to get cleaned after being released, so you raise a relevant point. I think it may be an issue that's out of scope for this issue, however.

**Comment by ljmcgann** [comment]
Yeah cleaned but not reinstantiated is what I meant. There needs to be a buffer from end_time to next start_time but I wont bother with that for now I guess

**Comment by ljmcgann** [comment]
Also, I will create a check to make sure start_time is actually before end_time. Seems like we need this
